src/LuaTransformer.ts

```diff
@@ -3895,6 +3895,14 @@ export class LuaTransformer {
         }
 
         switch (ownerType.flags) {
+            case ts.TypeFlags.Union:
```

Comment:
Would be much cleaner if you replaced this switch statement with a `tsHelper.isStringType` call (already exists). You can expand that function to deal with union/intersection types. Something like this:
```ts
export function isStringType(type: ts.Type): boolean {
    if (type.flags & ts.TypeFlags.Union) {
        return (type as ts.UnionType).types.every(isStringType);
    }

    if (type.flags & ts.TypeFlags.Intersection) {
        return (type as ts.IntersectionType).types.some(isStringType);
    }

    return (
        (type.flags & ts.TypeFlags.String) !== 0 ||
        (type.flags & ts.TypeFlags.StringLike) !== 0 ||
        (type.flags & ts.TypeFlags.StringLiteral) !== 0
    );
}
```

Reply:
There is a `ts.isUnionOrIntersection` function that can make it even simpler:

```ts
export function isStringType(type: ts.Type, checker: ts.TypeChecker, program: ts.Program): boolean {
    if (type.symbol) {
        const baseConstraint = checker.getBaseConstraintOfType(type);
        if (baseConstraint) {
            return isStringType(baseConstraint, checker, program);
        }
    }

    if (type.isUnionOrIntersection()) {
        return type.types.some(t => isStringType(t, checker, program));
    }

    return (
        (type.flags & ts.TypeFlags.String) !== 0 ||
        (type.flags & ts.TypeFlags.StringLike) !== 0 ||
        (type.flags & ts.TypeFlags.StringLiteral) !== 0
    );
}
```